api/client/samples/crop_models/brazil_soybeans.ipynb

```diff
@@ -8,7 +8,7 @@
     "In this notebook, we walk through the process of building a simple \"crop model\" using data in Gro. The accompanying Gro web app display for [Brazil Soybeans](https://app.gro-intelligence.com/displays/25894) provides an overview of the data series we will use. The crop model we will create is a very basic  model that uses crop-production-weighted NDVI to forecast  yield.\n",
```

Comment:
<p>In <code>compute_crop_weighted_series</code> we have:</p><p> </p><p> </p><p><code>for region in regions:</code></p><p><code>            entities['region_id'] = region['id']</code></p><p><code>            for data_series in self.get_data_series(**entities):</code></p><p><code>                self.add_single_data_series(data_series)</code></p><p><code>                break</code></p><p><code>df = self.get_df()</code></p><p> </p><p> </p><p>which could potentially add in series of varying units. To be safe, we could do something like grab the first unit_id from df and then apply self.convert_unit to each row to make sure they're consistent before getting the weights. All the resulting series are the same metric_id/item_id, so they should have the same set of allowed units.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/gro-intelligence/api-client/pull/153/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
actually it doesn't matter here.... since weighting is just applying a bunch of dimensionless constants across values in different regions, it will remain consistent regardless of units as long as units are the same across regions for a given point in time (don't even have to be same units across time or sources)

Reply:
Yeah it doesn't matter what the unit is - they just have to be consistent across the regions. But nothing here's guaranteeing that consistency.

You could easily get

```py
{ ..., 'region_id': 1215, 'source_id': 2, 'end_date': '2000-01-01', ''value': 1, 'unit_id': KG },
{ ..., 'region_id': 1216, 'source_id': 14, 'end_date': '2000-01-01', ''value': 1, 'unit_id': TONNES }
```

and then the weights output from `compute_weights` would be off.

Reply:
yes for compute_weights, ```weights = self.compute_weights(...) ``` assumes the weighting_crop_name, weighting_metric_name  series has consistent units across regions. let me add that in a separate PR since that is technically not part of the brazil soybeans

for the item_name and metric_name it doesn't matter since each value is independently has a weight applied to it (naturally the consumer might want to convert to whatever units)

Reply:
👍